src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -40,33 +52,56 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
   }
   async componentDidMount() {
     this._isMounted = true;
-    const selectedOption = this.props.selectedOption;
-    // early return if not possible to fetch data source
 
+    const selectedOption = this.props.selectedOption;
     const option = selectedOption[0];
     const optionId = option.id;
-    if (!option.label) {
-      try {
-        const title = (await getDataSourceById(optionId, this.props.savedObjectsClient)).title;
-        if (!title) {
-          this.props.notifications.addWarning(
-            i18n.translate('dataSource.fetchDataSourceError', {
-              defaultMessage: `Data source with id ${optionId} is not available`,
-            })
-          );
-        } else {
-          if (!this._isMounted) return;
-          this.setState({
-            selectedOption: [{ id: optionId, label: title }],
-          });
-        }
-      } catch (error) {
-        this.props.notifications.addWarning(
-          i18n.translate('dataSource.fetchDataSourceError', {
-            defaultMessage: `Failed to fetch data source due to ${error}`,
-          })
-        );
+
+    if (optionId === '' && !this.props.hideLocalCluster) {
+      this.setState({
+        selectedOption: [LocalCluster],
+      });
+      this.props.onSelectedDataSources([LocalCluster]);
+      return;
+    }
+
+    if (optionId === '' && this.props.hideLocalCluster) {
+      this.setState({
+        selectedOption: [],
+      });
+      this.props.onSelectedDataSources([]);
+      handleDataSourceFetchError(this.props.notifications!);
+      return;
+    }
+
+    if (
+      this.props.dataSourceFilter &&
+      this.state.selectedOption.filter(this.props.dataSourceFilter).length === 0
+    ) {
+      this.setState({
+        selectedOption: [],
+      });
+      this.props.onSelectedDataSources([]);
+      handleDataSourceFetchError(this.props.notifications!);
```

Comment:
This should be handleNoAvailableDataSourceError

Reply:
the function name is bit confusing, since we are not calling any function to fetch data source in this "if" block, there should be no error for us to handle right? Maybe `DisplayErrorToast`, `DisplayWarningToast` as a better naming?

Reply:
Let me rename the function. Thanks